prompb/io/prometheus/client/metrics.proto

```diff
@@ -51,6 +51,8 @@ message Gauge {
 message Counter {
   double   value    = 1;
   Exemplar exemplar = 2;
+
+  google.protobuf.Timestamp  created_timestamp = 3;
```

Comment:
This should be optional, but this deprecated library gogo-protobuf does not support optional fields 🙃

Reply:
There are lots of optional fields in the examples, e.g. https://github.com/gogo/protobuf/blob/f67b8970b736e53dbd7d0a27146c8f1ac52f74e5/test/example/example.proto#L50

Reply:
That's not a gogo vs. official protobuf problem, it's a proto2 vs proto3 problem.

proto3 doesn't distinguish between optional and required.

Reply:
When removing the optional, this is the [error I get in CI](https://github.com/prometheus/prometheus/actions/runs/6419467575/job/17429362731?pr=12936):

```
io/prometheus/client/metrics.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-gogofast hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.--gogofast_out: 
```

Reply:
Does 
```
      google.protobuf.Timestamp  created_timestamp = 3  [(gogoproto.nullable) = true];
```
Work?

Reply:
Neat. Would that maybe even help with [this issue](https://github.com/prometheus/prometheus/blob/a5a4eab679ccf2d432ab34b5143ef9e1c6482839/model/textparse/protobufparse.go#L150-L153)?

Reply:
In different news: Note that essentially everything is (formally) `optional` in the proto2 definition. And all of these fields are now "normal" proto3 fields. Same can be done for the created_timestamp. The problem is that then we cannot distinguish between "unset" and "set to default value", but maybe the suggested `[(gogoproto.nullable) = true]` helps with that.

About the possibility to set `optional` at all: I'm surprised that this is possible now (again). Here's the original rationale for not allowing `optional`: https://github.com/protocolbuffers/protobuf/issues/2497 (which is the same rational why all non-repeated fields are marked `optional` in proto2).

Apparently, `optional` was brought back in protobuf v3.15, with the aim to enable detection if a field was unset, which `[(gogoproto.nullable) = true]` should also allow. So I guess we'll go with the latter, and once we move on from gogoproto to another protobuf implementation (that is actually maintained), we can change this to `optional`.

Reply:
Noice, I wasn't aware that we could work around this way :) 

Thanks for the suggestion!

Reply:
Thanks everybody for helping!